Assets/MixedRealityToolkit.Services/InputSystem/GazePointerVisibilityStateMachine.cs

```diff
@@ -51,28 +51,6 @@ public void UpdateState(int numNearPointersActive, int numFarPointersActive, int
                 eyeGazeValid = isEyeGazeValid;
             }
 
-            if (isEyeGazeValid)
```

Comment:
Interesting, @wiwei and @sostel I remember you added this separate state machine for eye gaze to fix a specific bug, regarding eye tracking -- that hand interactions were not working with eyes. See #4482. With this change, can you still do near manipulations with eye tracking? Does this still pass the tests that will had added to verify that hands work with eyes?

Reply:
So it's been quite a while since I did that but from what I remember if eye gaze was enabled, if you had any hands in the scene that had active far/near pointers, it would disable eye gaze, and you had to say "select" to bring back eye gaze. It was essentially unusable from a user perspective.

It feels like this change seems to rip that out and bring it back to the old problem, but I also don't know if maybe the change to TryGetEyeGazeRay above is actually want makes it all work out. So maybe that actually is the better/right fix since it would enable gaze raycasting even if the eye gaze state isn't in the "active" state.

If https://github.com/microsoft/MixedRealityToolkit-Unity/issues/4431 isn't an issue with this code, then this change should be fine, though FWIW yeah there were some tests added to validate behavior, which should be changed to account for how cursors aren't visible now (but the ray itself should still be valid, etc)

Reply:
TryGetEyeGazeRay is trying to provide eye gaze data even if gaze is not eye based, but if that is the case hit position is not being handled, so all user have is ray direction. 
[Documentation](https://microsoft.github.io/MixedRealityToolkit-Unity/Documentation/EyeTracking/EyeTracking_TargetSelection.html#example-4-use-hand-rays-and-eye-gaze-input-together) says it is possible to use hands and eye gaze at the same time. Should this be achieved through the TryGetEyeGazeRay instead?

Reply:
> Should this be achieved through the TryGetEyeGazeRay instead?

Sorry but I'm not understanding what you mean here, could you explain a bit more?

Reply:
What I meant is that when eye is not selected as default gaze, if eye tracking data is available, the user can query the eye ray but If user wants to get focus information about that ray this is not implemented. 

Now I understand that this should be handled in a separate PR if needed.

Reply:
If I read it correctly, you removed handling automatically hiding the eye gaze cursor when the hand comes near a target. Where or how is this handled now? 
Also, why do you feel that ensuring a correctly returned eye gaze ray even if not set as primary input should be handled in a separate PR? I feel that it would fall under decoupling eye tracking data from eye-based control.

Reply:
> If I read it correctly, you removed handling automatically hiding the eye gaze cursor when > the hand comes near a target. Where or how is this handled now?

Now the logic for hiding gaze cursor when hand is visible is the same for both head or eye based. Hides when hand visible, unhides with "select" speech command. This is handled by FocusProvider and GazePointerVisibilityStateMachine.

> Also, why do you feel that ensuring a correctly returned eye gaze ray even if not set as primary input should be handled in a separate PR? I feel that it would fall under decoupling eye tracking data from eye-based control.

If eye is not set as gaze input, the eye gaze Ray can be queried from TryGetEyeGazeRay. If available, it will return the correct origin and direction of the eye ray. What cannot be queried is information about what object have eye focus, as FocusProvider is only handling gaze pointer from.

Reply:
This looks correct to me - We would want the same cursor behavior for eye gaze and head gaze, no? 
Also the data for querying the eye ray is available (InputRayUtils.TryGetEyeGaze) - the only thing that's missing is that the focus provider won't be able the tell you which object is focused with the eye ray in case the head gaze is active as we only have the target from the gaze pointer - we can add a follow up task for being able to query the object that is hit with the eye tracking ray in case we're in head gaze mode to have both of them available - but it's not something you would need eg for creating a heat map - for that the ray should be enough